pkg/controller/core/workload_controller.go

```diff
@@ -153,7 +156,15 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	}
 
 	if workload.SyncAdmittedCondition(&wl) {
-		return ctrl.Result{}, workload.ApplyAdmissionStatus(ctx, r.client, &wl, true)
+		if err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true); err != nil {
+			return ctrl.Result{}, err
+		}
+		if workload.IsAdmitted(&wl) {
+			if c := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadQuotaReserved); c != nil {
```

Comment:
is this check necessary? Can a workload have the Admitted condition, but not QuotaReserved? Looking here the workload will not get Admitted condition if the quota is not reserved: https://github.com/kubernetes-sigs/kueue/blob/615dc8e42d160174962f379dd8767e95298a1457/pkg/workload/admissionchecks.go#L51C7-L51C23.

Reply:
fair point, done